lib/iris/experimental/ugrid/__init__.py

```diff
@@ -3274,6 +3279,133 @@ def context(self):
 PARSE_UGRID_ON_LOAD = ParseUGridOnLoad()
 
 
+def _meshes_from_cf(cf_reader):
+    """
+    Common behaviour for extracting meshes from a CFReader.
+
+    Simple now, but expected to increase in complexity as Mesh sharing develops.
+
+    """
+    # Mesh instances are shared between file phenomena.
+    # TODO: more sophisticated Mesh sharing between files.
+    # TODO: access external Mesh cache?
+    mesh_vars = cf_reader.cf_group.meshes
+    meshes = {
+        name: _build_mesh(cf_reader, var, cf_reader.filename)
+        for name, var in mesh_vars.items()
+    }
+    return meshes
+
+
+def load_mesh(uris, var_name=None):
+    """
+    Create a single :class:`Mesh` object from one or more NetCDF files.
+
+    Raises an error if more/less than one :class:`Mesh` is found.
+
+    Parameters
+    ----------
+    uris : str or iterable of str
+        One or more filenames/URI's. Any URI's must support OpenDAP.
+    var_name : str, optional
+        Only return a :class:`Mesh` if its var_name matches this value.
+
+    Returns
+    -------
+    :class:`Mesh`
+
+    """
+    meshes_result = load_meshes(uris, var_name)
+    result = [mesh for file in meshes_result.values() for mesh in file]
+    mesh_count = len(result)
+    if mesh_count != 1:
+        message = (
+            f"Expecting 1 mesh, but input file(s) produced: {mesh_count} ."
+        )
+        raise ValueError(message)
+    return result[0]
+
+
+def load_meshes(uris, var_name=None):
+    """
+    Create :class:`Mesh` objects from one or more NetCDF files.
+
+    Parameters
+    ----------
+    uris : str or iterable of str
+        One or more filenames/URI's. Any URI's must support OpenDAP.
+    var_name : str, optional
+        Only return :class:`Mesh`'s that have var_names matching this value.
+
+    Returns
+    -------
+    dict
+        A dictionary of file paths/URL's and lists of the :class:`Mesh`'s
```

Comment:
Thanks, that's much nicer.
5ffc3d6606f71c0a1efde3d28fa36d25ad6bb4d9
7316ec62fbf40b07006814c0f1c99f39ef67ad59

Reply:
I would still say this ought to say "mapping each file path/URL **which contains :class:`Mesh`es** to a list ..."
Because otherwise it still sound like it should be 1-2-1 with the input  _which_ `_it_` is not.

Reply:
Sorry, just realised I already noted this in my #2 pass..

Reply:
b28fbd80bdb8007dc0c71831cd62e85756c5fc4f